### PR TITLE
update to purs 0.14.1, use spago, use github actions

### PR DESCRIPTION
**Pull request by srghma**

### CODE REVIEW

**Review by zyla — CHANGES_REQUESTED** [review]
This is super awesome, thanks!

There is one issue though: the new CI config doesn't run browser tests. Let's keep both CircleCI and GH Actions for now until this is sorted out.

File: test/node/DynamicSpec.purs [hunk 1]
```diff
@@ -278,40 +276,40 @@ spec = describe "Dynamic" $ do
...
-          append log x
+          -- | traceM $ "receiver 2: " <> show x
+          append log $ Left x
 
       liftEffect do
         inner <- newDynamic 1
         rootDynOuter.set inner
 
-      log `shouldHaveValue` [0, 1, 2]
+      log `shouldHaveValue` [(Right "receiver 1: 0"),(Left 0),(Right "set begin"),(Right "receiver 1: 1"),(Left 1),(Right "receiver 1: 2"),(Left 2),(Right "set end")]
```
[the remark this answers is not in the document]
Comment (srghma) [reply test/node/DynamicSpec.purs:302]: there was a warning (warnings are errors in github action tests) `you forgot to remove debug function`

so, I had to refactor)

**Comment by zyla** [comment]
> Let's keep both CircleCI and GH Actions for now until this is sorted out.

I'll merge this and fix the issue separately. Thanks for the contribution!

File: src/Specular/Dom/Builder.purs [hunk 2]
```diff
@@ -100,7 +100,7 @@ instance monadReplaceBuilder :: MonadReplace (Builder env) where
     liftEffect $ appendChild placeholderBefore env.parent
     liftEffect $ appendChild placeholderAfter env.parent
 
-    cleanupRef <- liftEffect $ newRef (mempty :: Effect Unit)
+    cleanupRef <- liftEffect $ new (mempty :: Effect Unit)
```
[the remark this answers is not in the document]
Comment (srghma) [reply src/Specular/Dom/Builder.purs:103]: TODO: replace everywhere

**Comment by srghma** [comment]
> There is one issue though: the new CI config doesn't run browser tests

hell, missed
yes, that's easy, we can also still remove CircleCI and leave only github actions

> I'll merge this and fix the issue separately. Thanks for the contribution

THX! I like people that are not like "this is my code, noone can touch it"!

**Pull request merged**